[pixman] build failure compiling both arm-windows and arm-uwp

Package: pixman[core]:arm-windows -> 0.40.0#4

**Host Environment**

- Host: x64-windows
- Compiler: MSVC 19.34.31933.0
-    vcpkg-tool version: 2022-11-10-5fdee72bc1fceca198fb1ab7589837206a8b81ba
    vcpkg-scripts version: 62d01b70d 2022-11-25 (28 hours ago)

**To Reproduce**

`vcpkg install --triplet arm-windows pixman`
`vcpkg install --triplet arm-uwp pixman`

**Failure logs**

```
-- Downloading https://www.cairographics.org/releases/pixman-0.40.0.tar.gz -> pixman-0.40.0.tar.gz...
-- Extracting source J:/vcpkg/downloads/pixman-0.40.0.tar.gz
-- Applying patch remove_test_demos.patch
-- Applying patch no-host-cpu-checks.patch
-- Applying patch fix_clang-cl.patch
-- Applying patch missing_intrin_include.patch
-- Using source at J:/vcpkg/buildtrees/pixman/src/0.40.0-f766b8cd75.clean
-- Found external ninja('1.11.0').
-- Getting CMake variables for arm-windows
-- Downloading https://www.python.org/ftp/python/3.10.7/python-3.10.7-embed-amd64.zip -> python-3.10.7-embed-amd64.zip...
-- Configuring arm-windows-dbg
-- Downloading https://repo.msys2.org/mingw/i686/mingw-w64-i686-pkg-config-0.29.2-3-any.pkg.tar.zst;https://www2.futureware.at/~nickoe/msys2-mirror/mingw/i686/mingw-w64-i686-pkg-config-0.29.2-3-any.pkg.tar.zst;https://mirror.yandex.ru/mirrors/msys2/mingw/i686/mingw-w64-i686-pkg-config-0.29.2-3-any.pkg.tar.zst;https://mirrors.tuna.tsinghua.edu.cn/msys2/mingw/i686/mingw-w64-i686-pkg-config-0.29.2-3-any.pkg.tar.zst;https://mirrors.ustc.edu.cn/msys2/mingw/i686/mingw-w64-i686-pkg-config-0.29.2-3-any.pkg.tar.zst;https://mirror.bit.edu.cn/msys2/mingw/i686/mingw-w64-i686-pkg-config-0.29.2-3-any.pkg.tar.zst;https://mirror.selfnet.de/msys2/mingw/i686/mingw-w64-i686-pkg-config-0.29.2-3-any.pkg.tar.zst;https://mirrors.sjtug.sjtu.edu.cn/msys2/mingw/i686/mingw-w64-i686-pkg-config-0.29.2-3-any.pkg.tar.zst -> msys-mingw-w64-i686-pkg-config-0.29.2-3-any.pkg.tar.zst...
-- Downloading https://repo.msys2.org/mingw/i686/mingw-w64-i686-libwinpthread-git-9.0.0.6373.5be8fcd83-1-any.pkg.tar.zst;https://www2.futureware.at/~nickoe/msys2-mirror/mingw/i686/mingw-w64-i686-libwinpthread-git-9.0.0.6373.5be8fcd83-1-any.pkg.tar.zst;https://mirror.yandex.ru/mirrors/msys2/mingw/i686/mingw-w64-i686-libwinpthread-git-9.0.0.6373.5be8fcd83-1-any.pkg.tar.zst;https://mirrors.tuna.tsinghua.edu.cn/msys2/mingw/i686/mingw-w64-i686-libwinpthread-git-9.0.0.6373.5be8fcd83-1-any.pkg.tar.zst;https://mirrors.ustc.edu.cn/msys2/mingw/i686/mingw-w64-i686-libwinpthread-git-9.0.0.6373.5be8fcd83-1-any.pkg.tar.zst;https://mirror.bit.edu.cn/msys2/mingw/i686/mingw-w64-i686-libwinpthread-git-9.0.0.6373.5be8fcd83-1-any.pkg.tar.zst;https://mirror.selfnet.de/msys2/mingw/i686/mingw-w64-i686-libwinpthread-git-9.0.0.6373.5be8fcd83-1-any.pkg.tar.zst;https://mirrors.sjtug.sjtu.edu.cn/msys2/mingw/i686/mingw-w64-i686-libwinpthread-git-9.0.0.6373.5be8fcd83-1-any.pkg.tar.zst -> msys-mingw-w64-i686-libwinpthread-git-9.0.0.6373.5be8fcd83-1-any.pkg.tar.zst...
-- Using msys root at J:/vcpkg/downloads/tools/msys2/9a1ec3f33446b195
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:96 (message):
    Command failed: J:/vcpkg/downloads/tools/python/python-3.10.7-x64/python.exe J:/vcpkg/installed/x64-windows/tools/meson/meson.py -Dmmx=disabled -Dsse2=disabled -Dssse3=disabled -Dlibpng=enabled --buildtype plain --backend ninja --wrap-mode nodownload --native J:/vcpkg/scripts/buildsystems/meson/none.txt --libdir lib --cross J:/vcpkg/buildtrees/pixman/meson-arm-windows-dbg.log -Ddebug=true --prefix J:/vcpkg/packages/pixman_arm-windows/debug --includedir ../include -Dcmake_prefix_path=['J:/vcpkg/installed/arm-windows/debug','J:/vcpkg/installed/arm-windows','J:/vcpkg/installed/arm-windows/share'] J:/vcpkg/buildtrees/pixman/src/0.40.0-f766b8cd75.clean
    Working Directory: J:/vcpkg/buildtrees/pixman/arm-windows-dbg
    Error code: 1
    See logs for more information:
      J:\vcpkg\buildtrees\pixman\config-arm-windows-dbg-meson-log.txt.log
      J:\vcpkg\buildtrees\pixman\config-arm-windows-dbg-out.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_configure_meson.cmake:432 (vcpkg_execute_required_process)
  ports/pixman/portfile.cmake:49 (vcpkg_configure_meson)
  scripts/ports.cmake:147 (include)



```
<details><summary>J:\vcpkg\buildtrees\pixman\config-arm-windows-dbg-meson-log.txt.log</summary>

```
Build started at 2022-11-26T14:32:48.675311
Main binary: J:\vcpkg\downloads\tools\python\python-3.10.7-x64\python.exe
Build Options: -Dmmx=disabled -Dsse2=disabled -Dssse3=disabled -Dlibpng=enabled -Ddebug=true '-Dcmake_prefix_path=['"'"'J:/vcpkg/installed/arm-windows/debug'"'"','"'"'J:/vcpkg/installed/arm-windows'"'"','"'"'J:/vcpkg/installed/arm-windows/share'"'"']' -Dprefix=J:/vcpkg/packages/pixman_arm-windows/debug -Dincludedir=../include -Dlibdir=lib -Dbackend=ninja -Dbuildtype=plain -Dwrap_mode=nodownload '--cross-file J:/vcpkg/buildtrees/pixman/meson-arm-windows-dbg.log' '--native-file J:/vcpkg/scripts/buildsystems/meson/none.txt'
Python system: Windows
The Meson build system
Version: 0.62.1
Source dir: J:\vcpkg\buildtrees\pixman\src\0.40.0-f766b8cd75.clean
Build dir: J:\vcpkg\buildtrees\pixman\arm-windows-dbg
Build type: cross build
Project name: pixman
Project version: 0.40.0
Sanity testing C compiler: C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.34.31933/bin/Hostx64/arm/cl.exe
Is cross compiler: True.
Sanity check compiler command line: C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.34.31933/bin/Hostx64/arm/cl.exe sanitycheckc.c /Fesanitycheckc_cross.exe -nologo -DWIN32 -D_WINDOWS -W3 -utf-8 -MP -D_DEBUG -MDd -Z7 -Ob0 -Od -RTC1 -IJ:/vcpkg/installed/arm-windows/include /nologo /showIncludes /utf-8 /c /link
Sanity check compile stdout:
sanitycheckc.c

-----
Sanity check compile stderr:

-----
C compiler for the host machine: C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.34.31933/bin/Hostx64/arm/cl.exe (msvc 19.34.31933 "Microsoft (R) C/C++ Optimizing Compiler Version 19.34.31933 for ARM")
C linker for the host machine: C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.34.31933/bin/Hostx64/arm/link.exe link 14.34.31933.0
Compiler for language c for the build machine not found.
Build machine cpu family: x86_64
Build machine cpu: x86_64
Host machine cpu family: arm
Host machine cpu: armv7hl
Target machine cpu family: arm
Target machine cpu: armv7hl
Running compile:
Working directory:  J:\vcpkg\buildtrees\pixman\arm-windows-dbg\meson-private\tmp86espw8s
Command line:  C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.34.31933/bin/Hostx64/arm/cl.exe -IJ:/vcpkg/installed/arm-windows/include J:\vcpkg\buildtrees\pixman\arm-windows-dbg\meson-private\tmp86espw8s\testfile.c /FoJ:\vcpkg\buildtrees\pixman\arm-windows-dbg\meson-private\tmp86espw8s\output.obj /nologo /showIncludes /utf-8 /c -nologo -DWIN32 -D_WINDOWS -W3 -utf-8 -MP -D_DEBUG -MDd -Z7 -Ob0 -Od -RTC1 /nologo /showIncludes /utf-8 /c /Od /Oi- -Wdeclaration-after-statement 

Code:
 extern int i;
int i;

Compiler stdout:
 
Compiler stderr:
 cl : Command line error D8021 : invalid numeric argument '/Wdeclaration-after-statement'

Compiler for C supports arguments -Wdeclaration-after-statement: NO 
Running compile:
Working directory:  J:\vcpkg\buildtrees\pixman\arm-windows-dbg\meson-private\tmpwk30_l79
...
Skipped 957 lines
...

Command line:  C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.34.31933/bin/Hostx64/arm/cl.exe -IJ:/vcpkg/installed/arm-windows/include J:\vcpkg\buildtrees\pixman\arm-windows-dbg\meson-private\tmpbistxfd6\testfile.c /FoJ:\vcpkg\buildtrees\pixman\arm-windows-dbg\meson-private\tmpbistxfd6\output.obj /nologo /showIncludes /utf-8 /c -nologo -DWIN32 -D_WINDOWS -W3 -utf-8 -MP -D_DEBUG -MDd -Z7 -Ob0 -Od -RTC1 /nologo /showIncludes /utf-8 /c /Od /Oi- 

Code:
 #include <stdio.h>
        
        int main(void) { static int a[1-2*!(sizeof(long) >= 0)]; a[0]=0; return 0; }
Compiler stdout:
 testfile.c
Note: including file: C:\Program Files (x86)\Windows Kits\10\include\10.0.22000.0\ucrt\stdio.h
Note: including file:  C:\Program Files (x86)\Windows Kits\10\include\10.0.22000.0\ucrt\corecrt.h
Note: including file:   C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.34.31933\include\vcruntime.h
Note: including file:    C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.34.31933\include\sal.h
Note: including file:     C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.34.31933\include\concurrencysal.h
Note: including file:    C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.34.31933\include\vadefs.h
Note: including file:  C:\Program Files (x86)\Windows Kits\10\include\10.0.22000.0\ucrt\corecrt_wstdio.h
Note: including file:   C:\Program Files (x86)\Windows Kits\10\include\10.0.22000.0\ucrt\corecrt_stdio_config.h

Compiler stderr:
 
Running compile:
Working directory:  J:\vcpkg\buildtrees\pixman\arm-windows-dbg\meson-private\tmpd50_1bq1
Command line:  C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.34.31933/bin/Hostx64/arm/cl.exe -IJ:/vcpkg/installed/arm-windows/include J:\vcpkg\buildtrees\pixman\arm-windows-dbg\meson-private\tmpd50_1bq1\testfile.c /FoJ:\vcpkg\buildtrees\pixman\arm-windows-dbg\meson-private\tmpd50_1bq1\output.obj /nologo /showIncludes /utf-8 /c -nologo -DWIN32 -D_WINDOWS -W3 -utf-8 -MP -D_DEBUG -MDd -Z7 -Ob0 -Od -RTC1 /nologo /showIncludes /utf-8 /c /Od /Oi- 

Code:
 #include <stdio.h>
        
        int main(void) { static int a[1-2*!(sizeof(long) > 0)]; a[0]=0; return 0; }
Compiler stdout:
 testfile.c
Note: including file: C:\Program Files (x86)\Windows Kits\10\include\10.0.22000.0\ucrt\stdio.h
Note: including file:  C:\Program Files (x86)\Windows Kits\10\include\10.0.22000.0\ucrt\corecrt.h
Note: including file:   C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.34.31933\include\vcruntime.h
Note: including file:    C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.34.31933\include\sal.h
Note: including file:     C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.34.31933\include\concurrencysal.h
Note: including file:    C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.34.31933\include\vadefs.h
Note: including file:  C:\Program Files (x86)\Windows Kits\10\include\10.0.22000.0\ucrt\corecrt_wstdio.h
Note: including file:   C:\Program Files (x86)\Windows Kits\10\include\10.0.22000.0\ucrt\corecrt_stdio_config.h

Compiler stderr:
 
Running compile:
Working directory:  J:\vcpkg\buildtrees\pixman\arm-windows-dbg\meson-private\tmp8m7hy40j
Command line:  C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.34.31933/bin/Hostx64/arm/cl.exe -IJ:/vcpkg/installed/arm-windows/include J:\vcpkg\buildtrees\pixman\arm-windows-dbg\meson-private\tmp8m7hy40j\testfile.c /FoJ:\vcpkg\buildtrees\pixman\arm-windows-dbg\meson-private\tmp8m7hy40j\output.obj /nologo /showIncludes /utf-8 /c -nologo -DWIN32 -D_WINDOWS -W3 -utf-8 -MP -D_DEBUG -MDd -Z7 -Ob0 -Od -RTC1 /nologo /showIncludes /utf-8 /c /Od /Oi- 

Code:
 #include <stdio.h>
        
        int main(void) { static int a[1-2*!(sizeof(long) > 1)]; a[0]=0; return 0; }
Compiler stdout:
 testfile.c
Note: including file: C:\Program Files (x86)\Windows Kits\10\include\10.0.22000.0\ucrt\stdio.h
Note: including file:  C:\Program Files (x86)\Windows Kits\10\include\10.0.22000.0\ucrt\corecrt.h
Note: including file:   C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.34.31933\include\vcruntime.h
Note: including file:    C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.34.31933\include\sal.h
Note: including file:     C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.34.31933\include\concurrencysal.h
Note: including file:    C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.34.31933\include\vadefs.h
Note: including file:  C:\Program Files (x86)\Windows Kits\10\include\10.0.22000.0\ucrt\corecrt_wstdio.h
Note: including file:   C:\Program Files (x86)\Windows Kits\10\include\10.0.22000.0\ucrt\corecrt_stdio_config.h

Compiler stderr:
 
Running compile:
Working directory:  J:\vcpkg\buildtrees\pixman\arm-windows-dbg\meson-private\tmpo6jhycbr
Command line:  C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.34.31933/bin/Hostx64/arm/cl.exe -IJ:/vcpkg/installed/arm-windows/include J:\vcpkg\buildtrees\pixman\arm-windows-dbg\meson-private\tmpo6jhycbr\testfile.c /FoJ:\vcpkg\buildtrees\pixman\arm-windows-dbg\meson-private\tmpo6jhycbr\output.obj /nologo /showIncludes /utf-8 /c -nologo -DWIN32 -D_WINDOWS -W3 -utf-8 -MP -D_DEBUG -MDd -Z7 -Ob0 -Od -RTC1 /nologo /showIncludes /utf-8 /c /Od /Oi- 

Code:
 #include <stdio.h>
        
        int main(void) { static int a[1-2*!(sizeof(long) > 3)]; a[0]=0; return 0; }
Compiler stdout:
 testfile.c
Note: including file: C:\Program Files (x86)\Windows Kits\10\include\10.0.22000.0\ucrt\stdio.h
Note: including file:  C:\Program Files (x86)\Windows Kits\10\include\10.0.22000.0\ucrt\corecrt.h
Note: including file:   C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.34.31933\include\vcruntime.h
Note: including file:    C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.34.31933\include\sal.h
Note: including file:     C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.34.31933\include\concurrencysal.h
Note: including file:    C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.34.31933\include\vadefs.h
Note: including file:  C:\Program Files (x86)\Windows Kits\10\include\10.0.22000.0\ucrt\corecrt_wstdio.h
Note: including file:   C:\Program Files (x86)\Windows Kits\10\include\10.0.22000.0\ucrt\corecrt_stdio_config.h

Compiler stderr:
 
Running compile:
Working directory:  J:\vcpkg\buildtrees\pixman\arm-windows-dbg\meson-private\tmpis02if_d
Command line:  C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.34.31933/bin/Hostx64/arm/cl.exe -IJ:/vcpkg/installed/arm-windows/include J:\vcpkg\buildtrees\pixman\arm-windows-dbg\meson-private\tmpis02if_d\testfile.c /FoJ:\vcpkg\buildtrees\pixman\arm-windows-dbg\meson-private\tmpis02if_d\output.obj /nologo /showIncludes /utf-8 /c -nologo -DWIN32 -D_WINDOWS -W3 -utf-8 -MP -D_DEBUG -MDd -Z7 -Ob0 -Od -RTC1 /nologo /showIncludes /utf-8 /c /Od /Oi- 

Code:
 #include <stdio.h>
        
        int main(void) { static int a[1-2*!(sizeof(long) > 7)]; a[0]=0; return 0; }
Compiler stdout:
 testfile.c
Note: including file: C:\Program Files (x86)\Windows Kits\10\include\10.0.22000.0\ucrt\stdio.h
Note: including file:  C:\Program Files (x86)\Windows Kits\10\include\10.0.22000.0\ucrt\corecrt.h
Note: including file:   C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.34.31933\include\vcruntime.h
Note: including file:    C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.34.31933\include\sal.h
Note: including file:     C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.34.31933\include\concurrencysal.h
Note: including file:    C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.34.31933\include\vadefs.h
Note: including file:  C:\Program Files (x86)\Windows Kits\10\include\10.0.22000.0\ucrt\corecrt_wstdio.h
Note: including file:   C:\Program Files (x86)\Windows Kits\10\include\10.0.22000.0\ucrt\corecrt_stdio_config.h
J:\vcpkg\buildtrees\pixman\arm-windows-dbg\meson-private\tmpis02if_d\testfile.c(3): error C2118: negative subscript

Compiler stderr:
 
Running compile:
Working directory:  J:\vcpkg\buildtrees\pixman\arm-windows-dbg\meson-private\tmp6bvl3sg8
Command line:  C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.34.31933/bin/Hostx64/arm/cl.exe -IJ:/vcpkg/installed/arm-windows/include J:\vcpkg\buildtrees\pixman\arm-windows-dbg\meson-private\tmp6bvl3sg8\testfile.c /FoJ:\vcpkg\buildtrees\pixman\arm-windows-dbg\meson-private\tmp6bvl3sg8\output.obj /nologo /showIncludes /utf-8 /c -nologo -DWIN32 -D_WINDOWS -W3 -utf-8 -MP -D_DEBUG -MDd -Z7 -Ob0 -Od -RTC1 /nologo /showIncludes /utf-8 /c /Od /Oi- 

Code:
 #include <stdio.h>
        
        int main(void) { static int a[1-2*!(sizeof(long) <= 5)]; a[0]=0; return 0; }
Compiler stdout:
 testfile.c
Note: including file: C:\Program Files (x86)\Windows Kits\10\include\10.0.22000.0\ucrt\stdio.h
Note: including file:  C:\Program Files (x86)\Windows Kits\10\include\10.0.22000.0\ucrt\corecrt.h
Note: including file:   C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.34.31933\include\vcruntime.h
Note: including file:    C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.34.31933\include\sal.h
Note: including file:     C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.34.31933\include\concurrencysal.h
Note: including file:    C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.34.31933\include\vadefs.h
Note: including file:  C:\Program Files (x86)\Windows Kits\10\include\10.0.22000.0\ucrt\corecrt_wstdio.h
Note: including file:   C:\Program Files (x86)\Windows Kits\10\include\10.0.22000.0\ucrt\corecrt_stdio_config.h

Compiler stderr:
 
Running compile:
Working directory:  J:\vcpkg\buildtrees\pixman\arm-windows-dbg\meson-private\tmpayd2mijp
Command line:  C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.34.31933/bin/Hostx64/arm/cl.exe -IJ:/vcpkg/installed/arm-windows/include J:\vcpkg\buildtrees\pixman\arm-windows-dbg\meson-private\tmpayd2mijp\testfile.c /FoJ:\vcpkg\buildtrees\pixman\arm-windows-dbg\meson-private\tmpayd2mijp\output.obj /nologo /showIncludes /utf-8 /c -nologo -DWIN32 -D_WINDOWS -W3 -utf-8 -MP -D_DEBUG -MDd -Z7 -Ob0 -Od -RTC1 /nologo /showIncludes /utf-8 /c /Od /Oi- 

Code:
 #include <stdio.h>
        
        int main(void) { static int a[1-2*!(sizeof(long) <= 4)]; a[0]=0; return 0; }
Compiler stdout:
 testfile.c
Note: including file: C:\Program Files (x86)\Windows Kits\10\include\10.0.22000.0\ucrt\stdio.h
Note: including file:  C:\Program Files (x86)\Windows Kits\10\include\10.0.22000.0\ucrt\corecrt.h
Note: including file:   C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.34.31933\include\vcruntime.h
Note: including file:    C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.34.31933\include\sal.h
Note: including file:     C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.34.31933\include\concurrencysal.h
Note: including file:    C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.34.31933\include\vadefs.h
Note: including file:  C:\Program Files (x86)\Windows Kits\10\include\10.0.22000.0\ucrt\corecrt_wstdio.h
Note: including file:   C:\Program Files (x86)\Windows Kits\10\include\10.0.22000.0\ucrt\corecrt_stdio_config.h

Compiler stderr:
 
Checking for size of "long" : 4
Configuring config.h using configuration
Configuring pixman-version.h using configuration
Compiler for C supports function attribute dllexport: YES 
WARNING: extract_all_objects called without setting recursive
keyword argument. Meson currently defaults to
non-recursive to maintain backward compatibility but
the default will be changed in the future.

WARNING: extract_all_objects called without setting recursive
keyword argument. Meson currently defaults to
non-recursive to maintain backward compatibility but
the default will be changed in the future.

Build targets in project: 3

pixman 0.40.0

  User defined options
    Cross files      : J:/vcpkg/buildtrees/pixman/meson-arm-windows-dbg.log
    Native files     : J:/vcpkg/scripts/buildsystems/meson/none.txt
    backend          : ninja
    buildtype        : plain
    cmake_prefix_path: ['J:/vcpkg/installed/arm-windows/debug','J:/vcpkg/installed/arm-windows','J:/vcpkg/installed/arm-windows/share']
    debug            : true
    includedir       : ../include
    libdir           : lib
    prefix           : J:/vcpkg/packages/pixman_arm-windows/debug
    wrap_mode        : nodownload
    libpng           : enabled
    mmx              : disabled
    sse2             : disabled
    ssse3            : disabled

Found ninja-1.11.0 at "C:/Program Files/Microsoft Visual Studio/2022/Community/Common7/IDE/CommonExtensions/Microsoft/CMake/Ninja\ninja.EXE"

ERROR: No specified compiler can handle file pixman\pixman-arm-simd-asm.S
```
</details>
<details><summary>J:\vcpkg\buildtrees\pixman\config-arm-windows-dbg-out.log</summary>

```
WARNING: Recommend using either -Dbuildtype or -Doptimization + -Ddebug. Using both is redundant since they override each other. See: https://mesonbuild.com/Builtin-options.html#build-type-options
The Meson build system
Version: 0.62.1
Source dir: J:\vcpkg\buildtrees\pixman\src\0.40.0-f766b8cd75.clean
Build dir: J:\vcpkg\buildtrees\pixman\arm-windows-dbg
Build type: cross build
Project name: pixman
Project version: 0.40.0
C compiler for the host machine: C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.34.31933/bin/Hostx64/arm/cl.exe (msvc 19.34.31933 "Microsoft (R) C/C++ Optimizing Compiler Version 19.34.31933 for ARM")
C linker for the host machine: C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.34.31933/bin/Hostx64/arm/link.exe link 14.34.31933.0
Compiler for language c for the build machine not found.
Build machine cpu family: x86_64
Build machine cpu: x86_64
Host machine cpu family: arm
Host machine cpu: armv7hl
Target machine cpu family: arm
Target machine cpu: armv7hl
Compiler for C supports arguments -Wdeclaration-after-statement: NO 
Compiler for C supports arguments -fno-strict-aliasing: NO 
Compiler for C supports arguments -fvisibility=hidden: NO 
Compiler for C supports arguments -Wundef: NO 
Compiler for C supports arguments -Wunused-local-typedefs: NO 
Checking if "ARMv6 SIMD Intrinsic Support" compiles: YES 
Checking if "NEON Intrinsic Support" compiles: YES 
Checking if "IWMMXT Intrinsic Support" compiles: NO 
Checking if "GNU Inline ASM support." compiles: NO 
Found pkg-config: J:/vcpkg/downloads/tools/msys2/9a1ec3f33446b195/mingw32/bin/pkg-config.exe (0.29.2)
Found CMake: J:/vcpkg/downloads/tools/cmake-3.24.0-windows/cmake-3.24.0-windows-i386/bin/cmake.exe (3.24.0)
Run-time dependency gtk+-2.0 found: NO (tried pkgconfig and cmake)
Run-time dependency glib-2.0 found: NO (tried pkgconfig and cmake)
Run-time dependency pixman-1 found: NO (tried pkgconfig and cmake)
Run-time dependency libpng found: YES 1.6.38
Library m found: NO
Run-time dependency threads found: YES
Has header "pthread.h" : NO 
Checking for function "sigaction" : NO 
Checking for function "alarm" : NO 
Checking for function "mprotect" : NO 
Checking for function "getpagesize" : NO 
Checking for function "mmap" : NO 
Checking for function "getisax" : NO 
Checking for function "gettimeofday" : NO 
Checking for function "feenableexcept" with dependency -lm: NO 
Header <fenv.h> has symbol "FE_DIVBYZERO" : YES 
Check usable header "sys/mman.h" : NO 
Check usable header "fenv.h" : YES 
Check usable header "unistd.h" : NO 
Compiler for C supports arguments -Werror=attributes: NO 
Checking if "TLS via __declspec(thread)" compiles: YES 
Checking if "__attribute__((constructor))" : links: NO 
Checking if "Has float128 support" : links: NO 
Checking for function "clz" : NO 
Checking if "Support for GCC vector extensions" : links: NO 
Checking for size of "long" : 4
Configuring config.h using configuration
Configuring pixman-version.h using configuration
Compiler for C supports function attribute dllexport: YES 
WARNING: extract_all_objects called without setting recursive
keyword argument. Meson currently defaults to
non-recursive to maintain backward compatibility but
the default will be changed in the future.

WARNING: extract_all_objects called without setting recursive
keyword argument. Meson currently defaults to
non-recursive to maintain backward compatibility but
the default will be changed in the future.

Build targets in project: 3

pixman 0.40.0

  User defined options
    Cross files      : J:/vcpkg/buildtrees/pixman/meson-arm-windows-dbg.log
    Native files     : J:/vcpkg/scripts/buildsystems/meson/none.txt
    backend          : ninja
    buildtype        : plain
    cmake_prefix_path: ['J:/vcpkg/installed/arm-windows/debug','J:/vcpkg/installed/arm-windows','J:/vcpkg/installed/arm-windows/share']
    debug            : true
    includedir       : ../include
    libdir           : lib
    prefix           : J:/vcpkg/packages/pixman_arm-windows/debug
    wrap_mode        : nodownload
    libpng           : enabled
    mmx              : disabled
    sse2             : disabled
    ssse3            : disabled

Found ninja-1.11.0 at "C:/Program Files/Microsoft Visual Studio/2022/Community/Common7/IDE/CommonExtensions/Microsoft/CMake/Ninja\ninja.EXE"

ERROR: No specified compiler can handle file pixman\pixman-arm-simd-asm.S

A full log can be found at J:\vcpkg\buildtrees\pixman\arm-windows-dbg\meson-logs\meson-log.txt
```
</details>

**Additional context**

The same error happens compiling both arm-windows and arm-uwp.
It seems that disabling both "arm-simd" and "neon" options may fix this issue.

--- a/ports/pixman/portfile.cmake
+++ b/ports/pixman/portfile.cmake
@@ -29,6 +29,12 @@ elseif(VCPKG_TARGET_ARCHITECTURE MATCHES "mips")
             -Dssse3=disabled)
 endif()
 
+if(VCPKG_TARGET_IS_UWP AND VCPKG_TARGET_ARCHITECTURE MATCHES "arm")
+   list(APPEND OPTIONS
+               -Darm-simd=disabled
+               -Dneon=disabled)
+endif()
+
 set(PIXMAN_VERSION 0.40.0)
 vcpkg_download_distfile(ARCHIVE
     URLS "https://www.cairographics.org/releases/pixman-${PIXMAN_VERSION}.tar.gz"

--- a/ports/pixman/portfile.cmake
+++ b/ports/pixman/portfile.cmake
@@ -29,7 +29,7 @@ elseif(VCPKG_TARGET_ARCHITECTURE MATCHES "mips")
             -Dssse3=disabled)
 endif()
 
-if(VCPKG_TARGET_IS_UWP AND VCPKG_TARGET_ARCHITECTURE MATCHES "arm")
+if(VCPKG_TARGET_IS_WINDOWS AND VCPKG_TARGET_ARCHITECTURE MATCHES "arm")
    list(APPEND OPTIONS
                -Darm-simd=disabled
                -Dneon=disabled)

--- a/ports/pixman/vcpkg.json
+++ b/ports/pixman/vcpkg.json
@@ -1,7 +1,7 @@
 {
   "name": "pixman",
   "version": "0.40.0",
-  "port-version": 4,
+  "port-version": 5,
   "description": "Pixman is a low-level software library for pixel manipulation, providing features such as image compositing and trapezoid rasterization.",
   "homepage": "https://www.cairographics.org/releases",
   "license": "MIT",

--- a/scripts/ci.baseline.txt
+++ b/scripts/ci.baseline.txt
@@ -799,7 +799,6 @@ pfring:arm64-osx=fail
 # pfring on Linux currently fails because its build scripts enable warnings as
 # errors, and warnings trigger with the Linux kernel headers in the Azure images.
 pfring:x64-linux=fail
-pixman:arm-uwp=fail
 platform-folders:arm-uwp=fail
 platform-folders:x64-uwp=fail
 plibsys:arm-uwp=fail

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -5794,7 +5794,7 @@
     },
     "pixman": {
       "baseline": "0.40.0",
-      "port-version": 4
+      "port-version": 5
     },
     "pkgconf": {
       "baseline": "1.8.0",

--- a/versions/p-/pixman.json
+++ b/versions/p-/pixman.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "f26230558cb33582de396b799e90a9046672b151",
+      "version": "0.40.0",
+      "port-version": 5
+    },
+    {
       "git-tree": "d0209cb23d5ca18cd74fa4a67e7ca80f7e81c0cd",
       "version": "0.40.0",
       "port-version": 4

